### ISSUE TITLE
Interpreter does not correctly interpret CREATE INDEX .. WHERE

### ISSUE THREAD

**Issue by lukaseder** [jOOQ/jOOQ#13013]
The following isn't interpreted correctly by our interpreter:

```sql
create table t (a int);
create index i on t (a) where a > 0;
```

The `WHERE` clause is being ignored. Likewise, this should throw, but doesn't:

```sql
create table t (b int);
create index i on t (b) where a > 0;
```

**Comment by lukaseder** [comment]
We're currently not validating the contents of a query (e.g. in views), or condition (e.g. in check constraints), so this still won't fail:

```sql
create index i on t (b) where a > 0;
```

**Comment by lukaseder** [comment]
Fixed in jOOQ 3.17.0, 3.16.4 (https://github.com/jOOQ/jOOQ/issues/13017), and 3.15.8 (https://github.com/jOOQ/jOOQ/issues/13018)
